### PR TITLE
Add support for SNS create and list subscriptions

### PR DESCRIPTION
**Pull request by taylorbrooks**

### CODE REVIEW

File: lib/ex_aws/sns.ex [hunk 1]
```diff
@@ -112,6 +112,23 @@ defmodule ExAws.SNS do
     |> Map.put(param_root <> ".Value.DataType", data_type)
   end
 
+  ## Subscriptions
+  ######################
+
+  @doc "Create Subscription"
+  def create_subscription(topic_arn, protocol, endpoint) do
```
Comment (benwilson512) [inline lib/ex_aws/sns.ex]: Typespecs please :)

File: lib/ex_aws/sns.ex [hunk 2]
```diff
@@ -112,6 +112,23 @@ defmodule ExAws.SNS do
...
+    request("Subscribe", %{
+      "TopicArn" => topic_arn,
+      "Protocol" => protocol,
+      "Endpoint" => endpoint,
+    })
+  end
+
+  @doc "List Subscriptions"
+  def list_subscriptions() do
```
Comment (benwilson512) [inline lib/ex_aws/sns.ex]: Typespecs here as well. This also needs to support the `NextToken` parameter covered here: http://docs.aws.amazon.com/sns/latest/api/API_ListSubscriptions.html

**Comment by benwilson512** [comment]
Hey there, happy to merge this if you make the changes from the review and fix merge issues.

**Comment by taylorbrooks** [comment]
Yep, I'll tackle it today.

**Comment by benwilson512** [comment]
Closing for now. Happy to merge if we can rebase and make reviewed changes.

**Pull request closed without merging**